### ISSUE TITLE
Social graph icons are not working/need improvements

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#623]
### Terms

- [X] I have searched all [open bug reports](https://github.com/activist-org/activist/issues?q=is%3Aopen+is%3Aissue+label%3Abug)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Behavior

At one point the social graph icons for activist were functioning, but they now appear to not be 🤔 I've installed the 1.0 version of Nuxt Devtools and will look into this as that includes a graph icon inspector.

**Comment by andrewtavis** [comment]
Something that I think might make sense to do for this is to look into [Nuxt SEO](https://nuxtseo.com/) for this. Using those modules will likely make a lot of this a lot easier.

**Comment by andrewtavis** [comment]
CC @GrandSuccess87 :)

**Comment by andrewtavis** [comment]
CC @arescrimson:

- The head file is at: https://github.com/activist-org/activist/blob/main/frontend/head.ts
- For local testing, you would be using the Nuxt dev tools
- First thing to do would be to be able to reflect the current state based on localhost

Also of consideration: https://github.com/activist-org/activist/issues/605

**Comment by IWang20** [comment]
Hi, I'm a bit unfamiliar with Nuxt before but I would love to try and learn/debug. Can I be assigned this issue?

**Comment by andrewtavis** [comment]
By all means you can be assigned, @IWang20! Thanks for your willingness to help out here :) This has been a pretty key issue for some time, so your support here would be very much appreciated 😊

**Comment by IWang20** [comment]
Where are the social graph icons are you referring to? 

![Image](https://github.com/user-attachments/assets/fd8e2da4-d82c-4a7e-8c4d-a742fed36658)

These are the only social media icons that I've seen so far on the homepage and the footer. 

I've also seen social media links on event/organization but I've just assumed those were placeholders.

**Comment by andrewtavis** [comment]
Thanks for checking in here, @IWang20 👋 By social graph icons we mean the icons that appear when the platform is shared on other social media platforms. So when you send an activist page on Signal, Telegram or another messenger we want an image to accompany the message. Similar for sharing on other social media platforms where there's an automatic image for the page. We need to configure this ourselves. As mentioned, we have [Nuxt SEO](https://nuxtseo.com/) that can help us with this, and from there we can use the Nuxt devtools to check the status of the social graph icons as there is functionality to check these built into it :)

Happy to share more and discuss further! Let me us know if you have questions on the above 😊

**Comment by IWang20** [comment]
<img src="https://github.com/user-attachments/assets/21cb9518-f97a-48b9-b3ae-c587e99edac1" width="700">
<br/>
Currently, nuxt devtools is not picking up on anything.
<br/>
According to Nuxt's tutorial on OG Images it's expecting a `defineOgImageComponent()` which I searched, isn't defined in the activist project <br/>
<img src="https://github.com/user-attachments/assets/3419f00d-02b2-48c8-8ee2-7833867204ce" width="500">

**Comment by andrewtavis** [comment]
That might be something for a first PR here then, @IWang20? Do you want to make a change and then I can check the devtools on your PR's branch? If you need an image of a certain dimension, then also let me know! We do have `activistOpenGraphImage.png` in `frontend/public/images/activist/` :)

**Comment by IWang20** [comment]
Sure, will get to that soon!

**Comment by IWang20** [comment]
I've been trying things out, and here's what I know so far.
[Nuxt OG Image - Getting Started](https://nuxtseo.com/docs/og-image/getting-started/getting-familar-with-nuxt-og-image)
Installing the package `nuxt-og-image` (via yarn) and adding it to `modules.ts` allows us to use built-in functions like `defineOgImageComponent()` where I can pass Vue templates and props to define `og:image`, `og:title`, and such. 
<br>
I ran into an issue, though. The instructions above say that `defineOgImageComponent()` is server-side only. I found that the function was only detectable by Nuxt after setting `ssr:true` in `nuxt.config.ts` and rerunning yarn dev. 
Server-side rendering is disabled by default, so I'm not sure how to approach this. 
![Image](https://github.com/user-attachments/assets/1934a5bc-27ee-4945-978f-2e319f9462c3)

**Comment by andrewtavis** [comment]
Thanks for looking into this, @IWang20 :) Do all the Nuxt SEO modules require SSR? We can revisit it being turned off if need be.

**Comment by IWang20** [comment]
No, I think that certain features within a module may require SSR. 
For instance, robots.txt can be dynamically generated, which requires SSR. [link](https://nuxtseo.com/learn/controlling-crawlers/robots-txt)
Other examples include build-time Route Rules for dynamic SEO meta tags [link](https://nuxtseo.com/docs/seo-utils/guides/route-rules), OG Images, and some more. But aside from OG Image, I think the rest should be features within modules that require SSR.

**Comment by andrewtavis** [comment]
Do we want to try to implement the features that don't require SSR first, and then we can move to those that do once we release and have a bit more capacity to focus on SSR as a main issue? :)

**Comment by IWang20** [comment]
That sounds good to me

**Comment by andrewtavis** [comment]
Unassigning for now, @IWang20 :) If you do want to pick this up again, then let us know!

**Comment by MayurK-cmd** [comment]
Hey Can i work on this?

**Comment by andrewtavis** [comment]
You'd be welcome to send along a solution to the issue, @MayurK-cmd, and then I'd assign and suggest that you open a PR from there 😊

**Comment by MayurK-cmd** [comment]
I would start by using Nuxt DevTools 1.0 to identify which components aren't rendering, then trace through the icon imports and check if there were recent changes after the Nuxt upgrade. Most likely causes: icon library not registered, assets missing from build, or CSS hiding them. I'll create a minimal reproduction, verify the fix, and add tests to prevent regression.

**Comment by andrewtavis** [comment]
Sounds good, @MayurK-cmd :) Feel free to look into this and report back on what you're finding. We'll then advise and you can open a PR 😊

**Comment by MayurK-cmd** [comment]
yes
